src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/ScoredDocument.java

```diff
@@ -14,22 +28,26 @@ public class ScoredDocument implements Comparable<ScoredDocument> {
 
   private final JsonNode id;
   private final ObjectNode document;
-  private final String passage;
+  private final Optional<String> passage;
   private DocumentScores scores;
 
-  ScoredDocument(JsonNode id, JsonNode document, String passage, DocumentScores scores) {
+  /** Package protected so we can test with explicit scores */
+  @VisibleForTesting
+  protected ScoredDocument(JsonNode id, JsonNode document, String passage, DocumentScores scores) {
     this(id, checkCast(document), passage, scores);
   }
 
   private static ObjectNode checkCast(JsonNode document) {
     if (document instanceof ObjectNode on) {
       return on;
     }
+    Objects.requireNonNull(document, "document cannot be null");
```

Comment:
I guess it's nicer in (very) unlikely case of getting non-ObjectNode, but simple 1-line cast would also work -- in rare fail case `ClassCastException` from specific line is easy to find. :-)